lib/solr/configuration.rb

```diff
@@ -1,16 +1,34 @@
+require 'solr/field'
+require 'solr/dynamic_field_type'
+require 'solr/field_definition_builder'
+require 'solr/type_definition_builder'
+
 module Solr
   class Configuration
-    attr_accessor :field_map, :open_timeout, :read_timeout, :url
+    attr_accessor :open_timeout, :read_timeout, :url, :field_types, :fields
 
     def initialize
-      @field_map = {}
       @read_timeout = 2
       @open_timeout = 8
-      @url = ENV.fetch('SOLR_URL')
+      @url = ENV['SOLR_URL']
```

Comment:
please make it a very explicit error, otherwise it will fail with a weird error when parsing the URI. 

Example:

if URL wasn't supplied (either implicitly or explicitly in the configuration), this error should be thrown:

```
Solrb gem requires you to set the URL of your Solr instance
either through SOLR_URL environmental variable or explicitly inside the configure block:
Solrb.configure do |config|
   config.url = 'http://localhost:8983/solr/core'
end
```

Tricky thing here is that user may not call the configure block and rely on implicit env var. 
So can be tricky to check the presence of a solr connection when loading the gem, but very easy to do it when actually trying to connect. Probably we should do it.